frontend/src/pages/user/shop/SellerCoupons.tsx

```diff
@@ -1,52 +1,97 @@
-import CouponItem from '@components/CouponItem';
 import { Col, Row } from 'react-bootstrap';
-import couponData from '@pages/coupon/couponData.json';
+import { useQuery } from '@tanstack/react-query';
+import { useNavigate } from 'react-router-dom';
+import { CheckFetchStatus, RouteOnNotOK } from '@lib/Status';
+import CouponItemShow from '@components/CouponItemShow';
 
-interface CouponProps {
-  id: number;
-  type: 'percentage' | 'fixed' | 'shipping';
-  name: string;
+interface ICoupon {
   description: string;
   discount: number;
-  start_date: string;
   expire_date: string;
-  tags: {
-    name: string;
-  }[];
+  id: number;
+  name: string;
+  scope: 'global' | 'shop';
+  start_date: string;
+  type: 'percentage' | 'fixed' | 'shipping';
 }
 
 const SellerCoupons = () => {
+  const navigate = useNavigate();
+
+  // TODO
+  // const {sellerName} = useParams();
+  const sellerName = 'user1';
+
+  const { data: CouponsData, status: fetchCouponsStatus } = useQuery({
+    queryKey: ['GetShopCoupons'],
+    queryFn: async () => {
+      const resp = await fetch(`/api/shop/${sellerName}/coupon?offset=0&limit=10`, {
+        method: 'GET',
+        headers: {
+          accept: 'application/json',
+        },
+      });
+      if (!resp.ok) {
+        RouteOnNotOK(resp, navigate);
+      } else {
+        return await resp.json();
+      }
+    },
+    select: (data) => data as ICoupon[],
+    enabled: true,
+    refetchOnWindowFocus: false,
+  });
+
+  if (fetchCouponsStatus !== 'success') {
+    return <CheckFetchStatus status={fetchCouponsStatus} />;
+  }
+
+  const globalCoupons = CouponsData.filter((coupon) => coupon.scope === 'global');
+  const shopCoupons = CouponsData.filter((coupon) => coupon.scope === 'shop');
```

Comment:
where can i set this? i didn't view it in `manageCoupons`
and i wonder what's the difference between them

Reply:
that's in [/sellerCoupon](http://localhost:8080/sellerID/coupons).
the fetched result will contain both admin and shop coupon, so i split them in differents at display.

Reply:
yes i know i can view it in `/sellerID/coupons` but i wonder how they being cataloged.
do you mean that if one is a admin, and this admin set the product, if will be put in global section, or just all global coupon reveals on every shop?

Reply:
all global coupon reveals on every shop.
for reminder, the ``all coupons`` page is deleted. this also means u can only see the global coupon in ``seller coupon``